css-view-transitions-2/Overview.bs

```diff
@@ -1079,55 +1084,63 @@ When capturing the old or new state for an element, perform the following steps
+</div>
 
-		1. Set |capturedElement|'s [=captured element/old transform=] to the transform that would map |element|’s [=border box=] from |containerElement|'s [=border box=] origin to its current visual position.
+### Compute the old 'view-transition-group' ### {#vt-group-capture-old}
+<div algorithm="additional old capture steps (group)">
+When  [[css-view-transitions-1#capture-old-state-algorithm|capturing the old state for an element]], perform the following steps given a [=captured element=] |capturedElement|, and an [=element=] |element|:
 
-		1. Set |capturedElement|'s [=captured element/containing group=] to |containingGroupName|.
+	1. Set |capturedElement|'s [=captured element/containing group name=] to the [=used value=] of |element|'s ''view-transition-group''.
 </div>
 
+### Compute the new 'view-transition-group' ### {#vt-group-capture-new}
 <div algorithm="additional new capture steps (group)">
 When [[css-view-transitions-1#capture-new-state-algorithm|capturing the new state for an element]], perform the following steps given a [=captured element=] |capturedElement| and an [=element=] |element|:
 
-	1. Set |capturedElement|'s [=captured element/containing group=] to the [=used value=] of |element|'s 'view-transition-group'.
+	1. Set |capturedElement|'s [=captured element/containing group name=] to the [=used value=] of |element|'s 'view-transition-group'.
 </div>
 
+### Reparent a ''::view-transition-group()'' to its specified containing group ### {#vt-group-reparent}
 <div algorithm="additional pseudo-element setup steps (group)">
-When [[css-view-transitions-1#setup-transition-pseudo-elements-algorithm|setting up the transition pseudo-element]] for a [=captured element=] |capturedElement|, given a |transitionName|:
+When [[css-view-transitions-1#setup-transition-pseudo-elements-algorithm|setting up the transition pseudo-element]] for a [=captured element=] |capturedElement|, given a |transitionName| and a |transition|:
+
+	1. Let |containingGroupName| be |capturedElement|'s [=containing group name=].
+	1. If |containingGroupName| is not null, then:
 
-	1. If |capturedElement|'s [=containing group=] is not null, then:
+		1. Let |groupContainerElement| be |transition|'s [=ViewTransition/named elements=][|containingGroupName|].
 
 		1. Let |group| be the ''::view-transition-group()'',
-			with its [=view transition =name=] set to |transitionName|.
+			whose [=view transition name=] is set to |transitionName|.
 
 		1. Let |parentGroup| be the ''::view-transition-group()'',
-			with its [=view transition name=] set to |capturedElement|'s [=containing group=].
+			whost [=view transition name=] is set to |containingGroupName|.
+
+		1. Set |capturedElement|'s [=captured element/old transform=] to the result of [=multiplying=] |capturedElement|'s [=captured element/old transform=] by the inverse matrix of |groupContainerElement|'s [=captured element/old transform=].
```

Comment:
IIUC we expect both the captured element's old transform and the groupContainerElement's old transform to be relative to the snapshot containing block here.

Wouldn't there be a bug for the case where we have A nesting B which nests C then? Since we build the pseudo tree top down, when we get to C, B's transform would already be relative to A but C's transform would be relative to the snapshot containing block.

Maybe we want to cache a "old transform from snapshot containing block" which is not mutated after capture?

Reply:
Fixed, not changing the old transform in the struct, only multiplying when setting the style.